lib/manager/npm/update/dependency/index.ts

```diff
@@ -1,8 +1,58 @@
 import { dequal } from 'dequal';
+import type { PackageJson } from 'type-fest';
 import { logger } from '../../../../logger';
 import { matchAt, replaceAt } from '../../../../util/string';
 import type { UpdateDependencyConfig } from '../../../types';
 
+function replaceAsString(
+  parsedContents: PackageJson,
+  fileContent: string,
+  depType: string,
+  depName: string,
+  oldVersion: string,
+  newValue: string
+): string | null {
+  // Update the file = this is what we want
+  // eslint-disable-next-line no-param-reassign
+  if (depName === oldVersion) {
+    delete Object.assign(parsedContents[depType], {
+      [newValue]: parsedContents[depType][oldVersion],
+    })[oldVersion];
```

Comment:
This looks strange to me, also coverage is missing.

Reply:
🤦 You're right. This is from my other work in #5558. I directly copy-pasted this function from there and didn't check more closely.